Enrich instructions for string functions HEX() and INSERT() (#16114)
pingcap/docs: functions-and-operators/string-functions.md

```sql
SELECT X'616263', HEX('abc'), UNHEX(HEX('abc')), 0x616263;
+-----------+------------+-------------------+----------+
| X'616263' | HEX('abc') | UNHEX(HEX('abc')) | 0x616263 |
+-----------+------------+-------------------+----------+
| abc       | 616263     | abc               | abc      |
+-----------+------------+-------------------+----------+
```

```sql
SELECT X'F09F8DA3', HEX('🍣'), UNHEX(HEX('🍣')), 0xF09F8DA3;
+-------------+-------------+--------------------+------------+
| X'F09F8DA3' | HEX('🍣')     | UNHEX(HEX('🍣'))     | 0xF09F8DA3 |
+-------------+-------------+--------------------+------------+
| 🍣            | F09F8DA3    | 🍣                   | 🍣           |
+-------------+-------------+--------------------+------------+
```

```sql
SELECT HEX(255), CONV(HEX(255), 16, 10);
+----------+------------------------+
| HEX(255) | CONV(HEX(255), 16, 10) |
+----------+------------------------+
| FF       | 255                    |
+----------+------------------------+
```

```sql
SELECT HEX(NULL);
+-----------+
| HEX(NULL) |
+-----------+
| NULL      |
+-----------+
```

### [`INSERT()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_insert)

> The `INSERT(str, pos, len, newstr)` function is used to replace a substring in `str` that starts at position `pos` and is `len` characters long with the string `newstr`. This function is multibyte safe.

Reviewer: ```suggestion
The `INSERT(str, pos, len, newstr)` function is used to replace a substring in `str` (that starts at position `pos` and is `len` characters long) with the string `newstr`. This function is multibyte safe.
```

### [`HEX()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_hex)

Return a hexadecimal representation of a decimal or string value.

> - For a string argument `HEX(str)` returns a hexademical string representation of `str` where each byte of each character in `str` is converted to two hexadecimal digits.

Reviewer 2: ```suggestion
- For a string argument `HEX(str)` returns a hexadecimal string representation of `str` where each byte of each character in `str` is converted to two hexadecimal digits. For example `a` in a UTF-8 or ASCII character set is encoded as a byte with a binary value of 00111101 or 61 in hexadecimal notation.
```
Feel free to include this or leave this out.

> - For a string argument `HEX(str)` returns a hexadecimal string representation of `str` where each byte of each character in `str` is converted to two hexadecimal digits. For example `a` in a UTF-8 or ASCII character set is encoded as a byte with a binary value of 00111101 or 61 in hexadecimal notation.

Reviewer 2: ```suggestion

- For a string argument `HEX(str)` returns a hexadecimal string representation of `str` where each byte of each character in `str` is converted to two hexadecimal digits. For example `a` in a UTF-8 or ASCII character set is encoded as a byte with a binary value of 00111101 or 61 in hexadecimal notation.
```

From the linter:
```
[2024-01-11T09:00:57.576Z] functions-and-operators/string-functions.md: this file has manual line breaks in the following lines:
[2024-01-11T09:00:57.576Z] 
[2024-01-11T09:00:57.576Z] MANUAL LINE BREAKS: L159
[2024-01-11T09:00:57.576Z] MANUAL LINE BREAKS: L197
[2024-01-11T09:00:57.576Z] MANUAL LINE BREAKS: L198
```

> - The inverse of this operation is performed by the [ `UNHEX()` ](https://docs.pingcap.com/tidb/dev/string-functions#unhex) function.

Reviewer 2: ```suggestion
- The inverse of this operation is performed by the [`UNHEX()`](/functions-and-operators/string-functions.md#unhex) function.
```

> - For a numeric argument `HEX(n)` returns a hexadecimal string representation of the value of `n` treated as a longlong (`BIGINT`) number. This is equivalent to [ `CONV(n, 10, 16)` ](https://docs.pingcap.com/tidb/stable/numeric-functions-and-operators#mathematical-functions).

Reviewer 2: ```suggestion
- For a numeric argument `HEX(n)` returns a hexadecimal string representation of the value of `n` treated as a longlong (`BIGINT`) number. This is equivalent to [ `CONV(n, 10, 16)` ](/functions-and-operators/numeric-functions-and-operators#mathematical-functions).
```

> - For a numeric argument `HEX(n)` returns a hexadecimal string representation of the value of `n` treated as a longlong (`BIGINT`) number. This is equivalent to [ `CONV(n, 10, 16)` ](/functions-and-operators/numeric-functions-and-operators#mathematical-functions).

Reviewer 2: ```suggestion
- For a numeric argument `HEX(n)` returns a hexadecimal string representation of the value of `n` treated as a longlong (`BIGINT`) number. This is equivalent to [`CONV(n, 10, 16)`](/functions-and-operators/numeric-functions-and-operators#mathematical-functions).
```

From the linter:
```
[2024-01-11T09:00:59.985Z] functions-and-operators/string-functions.md: 159: MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- For a string argument `HEX(s..."]
[2024-01-11T09:00:59.985Z] functions-and-operators/string-functions.md: 161: MD039/no-space-in-links Spaces inside link text [Context: "[ CONV(n, 10, 16) ]"]
[2024-01-11T09:00:59.985Z] functions-and-operators/string-functions.md: 198: MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Returs the original string `..."]
```
```

- For a `NULL` argument, this function returns `NULL`.

```sql

> SELECT X'616263', HEX('abc'), UNHEX(HEX('abc'));

Reviewer 2: Maybe add `0x616263`?
```
mysql> SELECT X'616263', HEX('abc'), UNHEX(HEX('abc')), 0x616263;
+-----------+------------+-------------------+----------+
| X'616263' | HEX('abc') | UNHEX(HEX('abc')) | 0x616263 |
+-----------+------------+-------------------+----------+
| abc       | 616263     | abc               | abc      |
+-----------+------------+-------------------+----------+
1 row in set (0.00 sec)
```

Author: Nice!
I add it as you said > https://github.com/pingcap/docs/pull/16114/commits/f4d74302a4857828117439221e4a7d38e264d632

> Return a hexadecimal representation of a decimal or string value.
> 
> - For a string argument `HEX(str)` returns a hexadecimal string representation of `str` where each byte of each character in `str` is converted to two hexadecimal digits. For example `a` in a UTF-8 or ASCII character set is encoded as a byte with a binary value of 00111101 or 61 in hexadecimal notation.
> - The inverse of this operation is performed by the [`UNHEX()`](/functions-and-operators/string-functions.md#unhex) function.
> - For a numeric argument `HEX(n)` returns a hexadecimal string representation of the value of `n` treated as a longlong (`BIGINT`) number. This is equivalent to [`CONV(n, 10, 16)`](/functions-and-operators/numeric-functions-and-operators#mathematical-functions).
> - For a `NULL` argument, this function returns `NULL`.

Reviewer 3: ```suggestion
The `HEX()` function is used to convert the given argument into a string representation of its hexadecimal value. The argument can be either a string or a number.

- If the argument is a string, `HEX(str)` returns a hexadecimal string representation of `str`. The function converts each byte of each character in `str` into two hexadecimal digits. For example, the character `a` in a UTF-8 or ASCII character set is represented as a binary value of `00111101`, or `61` in hexadecimal notation.
- If the argument is a number, `HEX(n)` returns a hexadecimal string representation of `n`. The function treats the argument `n` as a `BIGINT` number, equivalent to using `CONV(n, 10, 16)`.
- If the argument is `NULL`, the function returns `NULL`.
```

> ```sql
> SELECT X'616263', HEX('abc'), UNHEX(HEX('abc')), 0x616263;
> +-----------+------------+-------------------+----------+
> | X'616263' | HEX('abc') | UNHEX(HEX('abc')) | 0x616263 |
> +-----------+------------+-------------------+----------+
> | abc       | 616263     | abc               | abc      |
> +-----------+------------+-------------------+----------+
> 
> SELECT X'F09F8DA3', HEX('🍣'), UNHEX(HEX('🍣')), 0xF09F8DA3;
> +-------------+-------------+--------------------+------------+
> | X'F09F8DA3' | HEX('🍣')     | UNHEX(HEX('🍣'))     | 0xF09F8DA3 |
> +-------------+-------------+--------------------+------------+
> | 🍣            | F09F8DA3    | 🍣                   | 🍣           |
> +-------------+-------------+--------------------+------------+
> 
> SELECT HEX(255), CONV(HEX(255), 16, 10);
> +----------+------------------------+
> | HEX(255) | CONV(HEX(255), 16, 10) |
> +----------+------------------------+
> | FF       | 255                    |
> +----------+------------------------+
> 
> SELECT HEX(NULL);
> +-----------+
> | HEX(NULL) |
> +-----------+
> | NULL      |
> +-----------+
> ```

Reviewer 3: ````suggestion
Examples:

```sql
SELECT X'616263', HEX('abc'), UNHEX(HEX('abc')), 0x616263;
+-----------+------------+-------------------+----------+
| X'616263' | HEX('abc') | UNHEX(HEX('abc')) | 0x616263 |
+-----------+------------+-------------------+----------+
| abc       | 616263     | abc               | abc      |
+-----------+------------+-------------------+----------+
```

```sql
SELECT X'F09F8DA3', HEX('🍣'), UNHEX(HEX('🍣')), 0xF09F8DA3;
+-------------+-------------+--------------------+------------+
| X'F09F8DA3' | HEX('🍣')     | UNHEX(HEX('🍣'))     | 0xF09F8DA3 |
+-------------+-------------+--------------------+------------+
| 🍣            | F09F8DA3    | 🍣                   | 🍣           |
+-------------+-------------+--------------------+------------+
```

```sql
SELECT HEX(255), CONV(HEX(255), 16, 10);
+----------+------------------------+
| HEX(255) | CONV(HEX(255), 16, 10) |
+----------+------------------------+
| FF       | 255                    |
+----------+------------------------+
```

```sql
SELECT HEX(NULL);
+-----------+
| HEX(NULL) |
+-----------+
| NULL      |
+-----------+
```
````

### [`INSERT()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_insert)

> `INSERT(str, pos, len, newstr)` returns the string `str` with the substring beginning at position `pos` and `len` characters long which is replaced by `newstr`.
> 
> This function is multibyte safe.
> - Returs the original string `str` if `pos` is not within the length of string,
> - Returns the string which is replaced the rest of the string from position `pos` if `len` is not within the length of the rest of string.
> - Returns `NULL` if any argument is `NULL`.

Reviewer 3: ```suggestion
The `INSERT(str, pos, len, newstr)` function is used to replace a substring in `str` that starts at position `pos` and is `len` characters long with the string `newstr`. This function is multibyte safe.

- If `pos` exceeds the length of `str`, the function returns the original string `str` without modification.
- If `len` exceeds the remaining length of `str` from position `pos`, the function replaces the rest of the string from position `pos`.
- If any argument is `NULL`, the function returns `NULL`.
```

> ```sql
> SELECT INSERT('He likes tennis', 4, 5, 'plays');
> +------------------------------------------+
> | INSERT('He likes tennis', 4, 5, 'plays') |
> +------------------------------------------+
> | He plays tennis                          |
> +------------------------------------------+
> 
> SELECT INSERT('He likes tennis', -1, 5, 'plays');
> +-------------------------------------------+
> | INSERT('He likes tennis', -1, 5, 'plays') |
> +-------------------------------------------+
> | He likes tennis                           |
> +-------------------------------------------+
> 
> SELECT INSERT('He likes tennis', 4, 100, 'plays');
> +--------------------------------------------+
> | INSERT('He likes tennis', 4, 100, 'plays') |
> +--------------------------------------------+
> | He plays                                   |
> +--------------------------------------------+
> 
> SELECT INSERT('He likes tenis', 10, 100, '🍣');
> +-------------------------------------------+
> | INSERT('He likes tenis', 10, 100, '🍣')     |
> +-------------------------------------------+
> | He likes 🍣                                 |
> +-------------------------------------------+
> 
> SELECT INSERT('あああああああ', 2, 3, 'いいい');
> +----------------------------------------------------+
> | INSERT('あああああああ', 2, 3, 'いいい')           |
> +----------------------------------------------------+
> | あいいいあああ                                     |
> +----------------------------------------------------+
> 
> SELECT INSERT('あああああああ', 2, 3, 'xx');
> +---------------------------------------------+
> | INSERT('あああああああ', 2, 3, 'xx')        |
> +---------------------------------------------+
> | あxxあああ                                  |
> +---------------------------------------------+
> ```

Reviewer 3: ````suggestion
Examples:

```sql
SELECT INSERT('He likes tennis', 4, 5, 'plays');
+------------------------------------------+
| INSERT('He likes tennis', 4, 5, 'plays') |
+------------------------------------------+
| He plays tennis                          |
+------------------------------------------+
```

```sql
SELECT INSERT('He likes tennis', -1, 5, 'plays');
+-------------------------------------------+
| INSERT('He likes tennis', -1, 5, 'plays') |
+-------------------------------------------+
| He likes tennis                           |
+-------------------------------------------+
```

```sql
SELECT INSERT('He likes tennis', 4, 100, 'plays');
+--------------------------------------------+
| INSERT('He likes tennis', 4, 100, 'plays') |
+--------------------------------------------+
| He plays                                   |
+--------------------------------------------+
```

```sql
SELECT INSERT('He likes tenis', 10, 100, '🍣');
+-------------------------------------------+
| INSERT('He likes tenis', 10, 100, '🍣')     |
+-------------------------------------------+
| He likes 🍣                                 |
+-------------------------------------------+
```

```sql
SELECT INSERT('あああああああ', 2, 3, 'いいい');
+----------------------------------------------------+
| INSERT('あああああああ', 2, 3, 'いいい')           |
+----------------------------------------------------+
| あいいいあああ                                     |
+----------------------------------------------------+
```

```sql
SELECT INSERT('あああああああ', 2, 3, 'xx');
+---------------------------------------------+
| INSERT('あああああああ', 2, 3, 'xx')        |
+---------------------------------------------+
| あxxあああ                                  |
+---------------------------------------------+
```
````
